server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -174,9 +179,13 @@ private boolean assertInactiveWithNoKnownPeers() {
     }
 
     PeersResponse handlePeersRequest(PeersRequest peersRequest) {
```

Comment:
Should we run this on `CLUSTER_COORDINATION` rather than `GENERIC`? Likewise should we use `CLUSTER_COORDINATION` for the periodic wakeups, handling each `ProbeConnectionResult`, `SeedHostsResolver#resolveConfiguredHosts` and various other things that look to happen on `GENERIC` today?

Reply:
Yea I think so but I couldn't remember why I didn't use that pool here when I introduced it ... so I figured I wouldn't do anything that potentially broken for a simple PR like this one :)

Reply:
We run all this stuff on a single thread in `CoordinatorTests`, it should Just Work™.